### PR TITLE
#90 | Implementation of checking whether user has assignment for update user support status endpoint 

### PR DESCRIPTION
**Pull request by ibrahimdenizz**
# PULL REQUEST TEMPLATE

## Description

Implementation of checking whether user has assignment for update user support status endpoint.
When user update user support status to statuses that specified in issue, It checks whether user has assignment.
If user has assignment, it throws exception and doesn't update support status.

## Related Issue

#90 

## Proposed Changes

Please provide a brief and descriptive title for your pull request, summarizing the changes or features introduced. It
should be in English and clearly indicate the purpose of the pull request. Optionally, you can provide a more detailed
description of what you have done.

- Branches should follow the below `Naming Conventions of Branchs` header in [README.md](https://github.com/afet-yonetim-sistemi/ays-be/edit/main/README.md) File

## Additional Information

If there's any additional information or context that the reviewers should know, please provide it here.

## Checklist

Please check the following before submitting your pull request:

- [X] I have tested my changes locally and they are working as expected.
- [X] The code is formatted according to the project's coding guidelines and style.
- [X] Necessary documentation has been added or existing documentation has been updated.
- [X] Relevant unit tests have been written and included.
- [X] Relevant integration tests have been written and included.
- [X] The code has been reviewed to ensure its quality.
- [X] The code does not contain any issues flagged by SonarLint.
- [X] Default reviewers have been assigned to this pull request.
- [X] Assignees have been added if necessary.
- [X] Labels have been applied if necessary.
- [x] If an issue is opened and this pull request is related to that issue, it should not be associated with any project
  board or milestone. However, if there is no issue, the pull request should be associated with the appropriate project
  board (BE Board, In Review column) and milestone to track the progress.

## Reviewers

The default project maintainers and team members will be automatically assigned as reviewers for this pull request.

### CODE REVIEW

File: src/main/java/com/ays/user/service/impl/UserSelfServiceImpl.java [hunk 1]
```diff
@@ -34,6 +38,14 @@ public void updateUserSupportStatus(UserSupportStatusUpdateRequest updateRequest
         final UserEntity userEntity = userRepository.findById(userId)
                 .orElseThrow(() -> new AysUserNotExistByIdException(userId));
 
+        if (updateRequest.isSupportStatusChecked()) {
+            assignmentRepository
+                    .findByUserId(userId)
+                    .ifPresent(assignmentEntity -> {
+                        throw new AysUserAlreadyHasAssignmentException(userId, assignmentEntity.getId());
+                    });
+        }
```
Comment (agitrubard) [inline src/main/java/com/ays/user/service/impl/UserSelfServiceImpl.java:47]: Bu kısımda user'ı bulmaktan ziyade, şöyle bir logic yapmalıyız;
- eğer ki bu user'ın assignment'ı varsa
   - Assignment'ı varsa güncelleyebilececeği statülere göre bir kontrol yapılıp uygun exception fırlatılmalı
   - Assignment'ı yoksa standart bir şekilde güncelleyebileceği statüleri yine güncelleyebilmeli yani var olan akış değiştirilmemeli

File: src/main/java/com/ays/user/util/exception/AysUserAlreadyHasAssignmentException.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    /**
+     * Constructs a new {@code AysUserAlreadyPassiveException} with the specified id.
+     *
+     * @param id the id of the passive user
+     */
+    public AysUserAlreadyHasAssignmentException(String id, String assignmentId) {
+        super("USER ALREADY HAS ASSIGNMENT! id:" + id + " assignmentId:" + assignmentId);
+    }
+
+}
```
Comment (agitrubard) [inline src/main/java/com/ays/user/util/exception/AysUserAlreadyHasAssignmentException.java:27]: Java DOC'ların düzeltilmesi gerekiyor, sanırım güncellenmemiş

File: src/test/java/com/ays/assignment/model/entity/AssignmentEntityBuilder.java [hunk 3]
```diff
@@ -64,6 +69,16 @@ public AssignmentEntityBuilder withInstitution(InstitutionEntity institution) {
...
+    public AssignmentEntityBuilder withUserId(String userId) {
+        data.setUserId(userId);
+        return this;
+    }
+
+    public AssignmentEntityBuilder withUser(UserEntity user) {
+        data.setUser(user);
+        return this;
+    }
+
```
Comment (agitrubard) [inline src/test/java/com/ays/assignment/model/entity/AssignmentEntityBuilder.java:81]: Burada belki yapacağımız düzenlemeyle birlikte buradaki eklemlere ihtiyaç kalmayabilir

**Review by agitrubard — CHANGES_REQUESTED** [review]
Testleri geliştirmeler tamamlandıktan sonra tekrardan review ediyor olacam

File: src/main/java/com/ays/user/model/dto/request/UserSupportStatusUpdateRequest.java [hunk 4]
```diff
@@ -44,4 +44,15 @@ private boolean isSupportStatusAccepted() {
 
     }
 
+    public boolean isSupportStatusChecked() {
```
[the remark this answers is not in the document]
Comment (ibrahimdenizz) [reply src/main/java/com/ays/user/model/dto/request/UserSupportStatusUpdateRequest.java]: Güncellendi!

File: src/main/java/com/ays/user/util/exception/AysUserAlreadyHasAssignmentException.java (hunk 2 above)
Comment (ibrahimdenizz) [reply src/main/java/com/ays/user/util/exception/AysUserAlreadyHasAssignmentException.java:28]: Güncellendi!

File: src/main/java/com/ays/user/service/impl/UserSelfServiceImpl.java (hunk 1 above)
Comment (ibrahimdenizz) [reply src/main/java/com/ays/user/service/impl/UserSelfServiceImpl.java:74]: Güncellendi!

File: src/test/java/com/ays/assignment/model/entity/AssignmentEntityBuilder.java (hunk 3 above)
Comment (ibrahimdenizz) [reply src/test/java/com/ays/assignment/model/entity/AssignmentEntityBuilder.java:81]: Güncellendi!

**Review by agitrubard — APPROVED** [review]

**Review by anilhazar — APPROVED** [review]

**Pull request merged**